yarl/_url.py

```diff
@@ -261,6 +261,12 @@ def build(
         elif not user and not password and not host and not port:
             netloc = ""
         else:
+            if "@" in host and not user and not password:
```

Comment:
This seems like a weird place to extract this data from.. Could you add more detail regarding the problem you're trying to address?